src/js/components/jobs/shareModal.tsx

```diff
@@ -153,61 +292,8 @@ class ShareModal extends React.Component<WrappedProps> {
               <hr />
             </>
           ) : null}
-          <Input
-            className="slds-p-bottom_small"
-            id="share-job-link"
-            value={window.location.href}
-            type="url"
-            readOnly
-            fixedTextRight={
-              <Button
-                label={i18n.t('Copy Link')}
-                variant="brand"
-                onClick={this.handleCopy}
-                style={{ whiteSpace: 'nowrap' }}
-              />
-            }
-            inputRef={this.storeInputRef}
-            onFocus={this.handleFocus}
-          >
-            <div className="slds-form-element__help slds-text-color_success">
-              {transientMessageVisible ? i18n.t('Copied to clipboard') : ''}
-              {/* Space added to preserve height even when empty. */}
-              &nbsp;
-            </div>
-          </Input>
-
-          {job.user_can_edit ? (
-            <>
-              <RadioGroup
-                labels={{ label: i18n.t('Who can access this shared link?') }}
-                name="is_public"
-                onChange={this.handleChange}
-              >
-                <Radio
-                  id="is_public-false"
-                  label={i18n.t(
-                    'Only I and Salesforce staff can view this installation job.',
-                  )}
-                  value="false"
-                  checked={!job.is_public}
-                />
-                <Radio
-                  id="is_public-true"
-                  label={i18n.t(
-                    'Anyone with the link can view this installation job.',
-                  )}
-                  value="true"
-                  checked={job.is_public}
-                />
-              </RadioGroup>
-              <p className="slds-text-body_small slds-p-top_small">
-                {i18n.t(
-                  'Access to view the installation job does not provide access to your Salesforce org.',
-                )}
-              </p>
-            </>
-          ) : null}
+          {this.getScratchOrgInfo()}
+          {this.getShareForm()}
```

Comment:
It seems weird to have a component called ShareModal when sharing is only one of the two things it's used for. What's the motivation for making this be one component that toggles between different contents, instead of 2 different components?

Reply:
@davisagli Yeah, I went back and forth on this. In the end, both cases are using the same copy-URL "share link", which led me to leave these combined. The differences are around the intro text, footer actions, and whether or not we also include the "make it public" radios. I thought it ended up cleaner and clearer to keep them together in this case?

Reply:
I don't feel strongly about it.